test/client/http_service.test.js

```diff
@@ -0,0 +1,265 @@
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.put(TEST_SERVER_URL);
+
+        expect(statusCode).toBe(204);
+        expect(body).toBeFalsy();
+      });
+
+      it('with options should work (raw body)', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.put({
+          url: TEST_SERVER_URL,
+          body: { foo: 'bar' },
+          encodeBodyAsJSON: true
+        });
+
+        expect(statusCode).toBe(204);
+        expect(body).toBeFalsy();
+      });
+
+      it('with options should work (stringified body)', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.put({
+          url: TEST_SERVER_URL,
+          body: JSON.stringify({ foo: 'bar' })
+        });
+
+        expect(statusCode).toBe(204);
+        expect(body).toBeFalsy();
+      });
+    });
+
+    describe.skip('Submitting a PATCH request', () => {
+      const requestHandler = async(req, res) => {
+        const body = await readRequestBody(req);
+        res.writeHead(206, { 'Content-Type': 'text/plan' });
+        res.end(body);
+      };
+
+      it('with url should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.patch(TEST_SERVER_URL);
+
+        expect(statusCode).toBe(206);
+        expect(body).toBeFalsy();
+      });
+
+      it('with options should work (raw body)', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.patch({
+          url: TEST_SERVER_URL,
+          body: { foo: 'bar' },
+          encodeBodyAsJSON: true
+        });
+
+        expect(statusCode).toBe(206);
+        expect(body).toBeFalsy();
+      });
+
+      it('with options should work (stringified body)', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.patch({
+          url: TEST_SERVER_URL,
+          body: JSON.stringify({ foo: 'bar' })
+        });
+
+        expect(statusCode).toBe(206);
+        expect(body).toBeFalsy();
+      });
+    });
+
+    describe('Submitting a DELETE request', () => {
+      const requestHandler = (req, res) => {
+        res.writeHead(202, { 'Content-Type': 'text/plan' });
+        res.end();
+      };
+
+      it('with url should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.delete(TEST_SERVER_URL);
+
+        expect(statusCode).toBe(202);
+        expect(body.toString()).toBe('');
+      });
+
+      it('with options should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.delete({ url: TEST_SERVER_URL });
+
+        expect(statusCode).toBe(202);
+        expect(body.toString()).toBe('');
+      });
+    });
+
+    describe.skip('Submitting a HEAD request', () => {
```

Comment:
is the `.skip` meant to be here?

Reply:
yeah, refer to the comment [above](https://github.com/mongodb/stitch-js-sdk/pull/90#discussion_r154365661)